_posts/en/newsletters/2020-07-29-newsletter.md

```diff
@@ -0,0 +1,192 @@
+projects.
+
+## Action items
+
+*None this week.*
+
+## News
+
+- **Upgrading channel commitment formats:** Olaoluwa Osuntokun
+  [posted][osuntokun upgrade] to the Lightning-Dev mailing list this
+  week with a suggested extension to the LN specification that will
+  allow two peers with an existing channel between them to negotiate a
+  new commitment transaction using a different format.  Commitment
+  transactions are used to allow LN nodes to put the current channel
+  state on chain, but the existing protocol only allows nodes to upgrade
+  to new commitment formats by opening new channels.  Osuntokun's
+  proposal would allow a node to signal to its peer that it wants to
+  switch formats.  If the peer agrees, the two nodes would port their
+  existing channel state over to the new format and then use the new
+  format going forward.
+
+    All discussion participants seemed to support the basic idea.
+    Bastien Teinturier [suggested][teinturier simple] that
+    it would be simplest to only allow switching commitment formats when
+    channels had no pending payments (HTLCs)---implying nodes would
+    need to pause sending or relaying payments in a particular channel
+    in order to upgrade it.
+
+    ZmnSCPxj [noted][zmnscpxj re-funding] that the same basic idea could
+    be used to essentially update the funding transaction offchain, such
+    as the case where [taproot][topic taproot]
+    and [SIGHASH_ANYPREVOUT][topic sighash_noinput] are implemented,
+    allowing [Eltoo][topic eltoo]-based channel commitments to be used.  In
+    ZmnSCPxj's proposal, the output of the existing funding transaction
+    would be paid to a new funding transaction that is kept offchain.  If
+    the channel terminates with a mutual close, the original funding
+    transaction output is paid to the final channel balances; otherwise,
+    the offchain secondary funding transaction can be published onchain
+    and the channel can be resolved using the appropriate unilateral
+    close protocol.
+
+## Field report: Using Descriptors and PSBT at River
+
+{% include articles/river-descriptors-psbt.md %}
+
+## Selected Q&A from Bitcoin StackExchange
+
+*[Bitcoin StackExchange][bitcoin.se] is one of the first places Optech
+contributors look for answers to their questions---or when we have a
+few spare moments to help curious or confused users.  In
+this monthly feature, we highlight some of the top-voted questions and
+answers posted since our last update.*
+
+{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [What are "leaf versions" in Taproot?]({{bse}}97104)
+  Michael Folkson and Pieter Wuille explain that each leaf node of a
+  [taproot][topic taproot] tree has its own leaf version in addition to the
+  spending script. This leaf version can be used for upgradability and
```

Comment:
I think some Bitcoin writers have previously used the term "spending script" to mean the scriptSig, what me might call today the witness data.  The data included in the taproot leaves is much more akin to what those writers sometimes called the "locking script".  I think here we can just be a bit vague and simply call it "the script".  Maybe: "each leaf node of a taproot tree commits to both a leaf version and a script."